test/integration/controller_test.go

```diff
@@ -97,12 +98,7 @@ func TestBasicFlows(t *testing.T) {
 		},
 		ProvisionReaction: &fakeosb.ProvisionReaction{
 			Response: &osb.ProvisionResponse{
-				Async: true,
-			},
-		},
-		PollLastOperationReaction: &fakeosb.PollLastOperationReaction{
-			Response: &osb.LastOperationResponse{
-				State: osb.StateSucceeded,
+				Async: false,
```

Comment:
For the record, I'm not even convinced this PR actually changed things - I just happened to get flakes in CI that I happened to be able to consistently (maybe 40% of runs) reproduce in my local system when debugging.